File: 013/main.go
package main

import (
	"io"
	"log"
	"net/http"
	"strings"
	"time"

	"github.com/siuyin/dflt"
)

func main() {
	http.Handle("/", http.FileServer(http.Dir("./html")))
	http.HandleFunc("/md", mdHandler)

	port := dflt.EnvString("PORT", "8080")
	log.Printf("Starting streaming markdown endpoint /md on port %s.\n", port)
	log.Fatal(http.ListenAndServe(":"+port, nil))
}

func mdHandler(w http.ResponseWriter, r *http.Request) {
	txt := `# This is markdown
The quick brown fox jumps over the lazy dog.  
The rain in spain falls mainly on the plain.  
A stitch in time saves nine.  
The cat sat on the mat.

Next we have an ordered list:
1. a
1. b
1. c

## Next header
And a bullet list:
* apples
* bananas
* cranberries
`
	p := strings.Split(txt, " ")
	w.Header().Set("Transfer-Encoding", "chunked")
	f, _ := w.(http.Flusher)
	for _, wd := range p {
		io.WriteString(w, wd+" ")
		time.Sleep(150 * time.Millisecond)
		f.Flush()
	}
}

File: 013/main.js
import { serveDir } from "@std/http/file-server"
import { EnvString } from "dflt"
function main() {
    function mainHandler(req) {
        const path = new URL(req.url).pathname;

        if (path === "/md") {
            return mdHandler(req)
        }

        return serveDir(req, {
            fsRoot: "html",
            urlRoot: "",
        });
    }
    const port = EnvString("PORT", "8080")
    console.log(`Starting streaming markdown endpoint /md on port: ${port}.`)
    Deno.serve({ port: port }, mainHandler)
}
main()


function mdHandler(req) {
    const hdrs = new Headers()
    hdrs.set("Transfer-Encoding", "chunked")
    hdrs.set("Access-Control-Allow-Origin", "*")

    const res = new Response("markdown here", {headers:hdrs})
    return res
}